lib/processors/minifier.js

```diff
@@ -86,20 +160,69 @@ export default async function({resources, taskUtil, options: {addSourceMappingUr
 	}
 
 	return Promise.all(resources.map(async (resource) => {
-		const dbgPath = resource.getPath().replace(debugFileRegex, "-dbg$1");
-		const dbgResource = await resource.clone();
-		dbgResource.setPath(dbgPath);
+		const resourcePath = resource.getPath();
+		const dbgPath = resourcePath.replace(debugFileRegex, "-dbg$1");
+		const dbgFilename = posixPath.basename(dbgPath);
 
 		const filename = posixPath.basename(resource.getPath());
-		const code = await resource.getString();
 
 		const sourceMapOptions = {
 			filename
 		};
 		if (addSourceMappingUrl) {
 			sourceMapOptions.url = filename + ".map";
 		}
-		const dbgFilename = posixPath.basename(dbgPath);
+
+		// Extract *and remove* source map reference from resource before cloning it
+		const sourceMappingUrl = await extractAndRemoveSourceMappingUrl(resource);
+
+		const code = await resource.getString();
+		// Create debug variant based off the original resource before minification
+		const dbgResource = await resource.clone();
+		dbgResource.setPath(dbgPath);
+
+		let dbgSourceMapResource;
+		if (sourceMappingUrl) {
+			if (resource.getSourceMetadata()?.contentModified) {
```

Comment:
This flag becomes `true` via `resource.setString` in `extractAndRemoveSourceMappingUrl`, which doesn't fit to what this check is supposed to do.

This doesn't appear in the current tests as the resources don't have sourceMetadata, thus the flag is not set at all. This applies to resources created by tasks, e.g. the transpile task.

But when files are already part of the project sources (i.e. transpiled upfront outside of ui5 build), the `contentModified` flag will always be `true`, so source maps will be ignored.

Reply:
Fixed via https://github.com/SAP/ui5-builder/pull/780/commits/c87f8834f3076db90f8156a09a1757f3eba82f05